### PR TITLE
Update openshift_cert_expiry for py3 support.

### LINKED ISSUE #4934: AttributeError: 'str' object has no attribute 'decode'
#### Description

When attempting to redeploy a new CA I get an error.

The initial ansible cluster installation went fine.

```
TASK [openshift_certificate_expiry : Check cert expirys on host] ***********************************************************************************************************
Friday 28 July 2017  16:08:21 +0000 (0:00:00.031)       0:00:49.617 *********** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'str' object has no attribute 'decode'
fatal: [router1]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_ce68sq3o/ansible_module_openshift_cert_expiry.py\", line 858, in <module>\n    main()\n  File \"/tmp/ansible_ce68sq3o/ansible_module_openshift_cert_expiry.py\", line 589, in main\n    cert_serial) = load_and_handle_cert(c, now, base64decode=True, ans_module=module)\n  File \"/tmp/ansible_ce68sq3o/ansible_module_openshift_cert_expiry.py\", line 282, in load_and_handle_cert\n    _cert_string = cert_string.decode('base-64')\nAttributeError: 'str' object has no attribute 'decode'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'str' object has no attribute 'decode'
fatal: [master1]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_sah7j28y/ansible_module_openshift_cert_expiry.py\", line 858, in <module>\n    main()\n  File \"/tmp/ansible_sah7j28y/ansible_module_openshift_cert_expiry.py\", line 619, in main\n    cert_serial) = load_and_handle_cert(c, now, base64decode=True, ans_module=module)\n  File \"/tmp/ansible_sah7j28y/ansible_module_openshift_cert_expiry.py\", line 282, in load_and_handle_cert\n    _cert_string = cert_string.decode('base-64')\nAttributeError: 'str' object has no attribute 'decode'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
```

##### Version

```
openshift-ansible-3.5.102-1

ansible 2.3.1.0
  config file = /home/centos/openshift/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

```
oc v1.5.1+7b451fc
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

openshift v1.5.1+7b451fc
kubernetes v1.5.2+43a9be4

```

##### Steps To Reproduce
1. `ansible-playbook -i hosts openshift-ansible/playbooks/byo/openshift-cluster/redeploy-openshift-ca.yml`  on an existing cluster

##### Additional Information

Bastion is CentOS7 (python2)

Cluster members are Fedora 26 Atomic (python3)

### PR DESCRIPTION
**Pull request by abutcher**
Fixes #4934

### CODE REVIEW

File: roles/openshift_certificate_expiry/library/openshift_cert_expiry.py [hunk 1]
```diff
@@ -272,15 +247,15 @@ def load_and_handle_cert(cert_string, now, base64decode=False, ans_module=None):
...
+- `base64decode` (bool) - run base64.b64decode() on the input?
 - `ans_module` (AnsibleModule) - The AnsibleModule object for this module (so we can raise errors)
 
 Returns:
 A tuple of the form:
     (cert_subject, cert_expiry_date, time_remaining, cert_serial_number)
     """
     if base64decode:
-        _cert_string = cert_string.decode('base-64')
+        _cert_string = base64.b64decode(cert_string).decode('utf-8')
```
Comment (ashcrow) [inline roles/openshift_certificate_expiry/library/openshift_cert_expiry.py]: 👍

File: roles/openshift_certificate_expiry/library/openshift_cert_expiry.py [hunk 2]
```diff
@@ -310,6 +285,9 @@ def load_and_handle_cert(cert_string, now, base64decode=False, ans_module=None):
     # Read all possible names from the cert
     cert_subjects = []
     for name, value in cert_loaded.get_subject().get_components():
+        if isinstance(name, bytes) or isinstance(value, bytes):
```
Comment (ashcrow) [inline roles/openshift_certificate_expiry/library/openshift_cert_expiry.py]: Should work fine.

File: tox.ini [hunk 3]
```diff
@@ -24,4 +24,4 @@ commands =
     generate_validation: python setup.py generate_validation
     # TODO(rhcarvalho): check syntax of other important entrypoint playbooks
     ansible_syntax: python setup.py ansible_syntax
-    integration: python -c 'print "run test/integration/run-tests.sh"'
+    integration: python -c 'print("run test/integration/run-tests.sh")'
```
Comment (ashcrow) [inline tox.ini]: 👍

File: roles/openshift_certificate_expiry/library/openshift_cert_expiry.py [hunk 4]
```diff
@@ -272,15 +247,15 @@ def load_and_handle_cert(cert_string, now, base64decode=False, ans_module=None):
 
 - `cert_string` (string) - a certificate loaded into a string object
 - `now` (datetime) - a datetime object of the time to calculate the certificate 'time_remaining' against
-- `base64decode` (bool) - run .decode('base64') on the input?
+- `base64decode` (bool) - run base64.b64decode() on the input?
```
Comment (ashcrow) [inline roles/openshift_certificate_expiry/library/openshift_cert_expiry.py]: nit: Why is this a question?

Comment (abutcher) [inline roles/openshift_certificate_expiry/library/openshift_cert_expiry.py]: We're now certain of this.

File: roles/openshift_certificate_expiry/library/openshift_cert_expiry.py (hunk 2 above)
Comment (abutcher) [inline roles/openshift_certificate_expiry/library/openshift_cert_expiry.py]: Yeah, this is the one I'm least happy about. I get bytes in py3 here.

**Comment by abutcher** [comment]
aos-ci-test

**Comment by sdodson** [comment]
[merge]

**Comment by tbielawa** [comment]
* https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/785/

Is a noted recent CI failure per @stevekuznetsov 

> Your job failed from this issue if it has:
>
> Stage ENSURE BUILT VERSION OF ORIGIN IS INSTALLED [00h 00m 05s] failed.

* https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible/395/

Suggests a network failure 

```
Aug  2 18:53:32.541: INFO: At 2017-08-02 18:50:13 -0400 EDT - event for liveness-http: {kubelet ip-172-18-6-97.ec2.internal} Unhealthy: Liveness probe failed: Get http://10.128.0.104:80/: dial tcp 10.128.0.104:80: getsockopt: no route to host
```

Unsure of how that could be related to the changes in this PR.

**Comment by tbielawa** [comment]
re[test]

**Comment by tbielawa** [comment]
[merge]

**Comment by sdodson** [comment]
[merge]

**Pull request merged**